src/main/java/org/cactoos/text/TextOf.java

```diff
@@ -384,276 +354,13 @@ public TextOf(final InputStream input) {
-                )
-            )
-        );
-    }
-
-    /**
-     * Formats date using ISO date time format.
-     * @param date The date to format.
-     */
-    public TextOf(final OffsetDateTime date) {
-        this(date, new Iso().value());
-    }
-
-    /**
-     * Formats date using provided date time format string using default locale.
-     * @param date The date to format.
-     * @param format The format to use.
-     */
-    public TextOf(final OffsetDateTime date, final String format) {
-        this(date, format, Locale.getDefault(Locale.Category.FORMAT));
-    }
-
-    /**
-     * Formats the date using the provided format string using the provided
-     * locale.
-     * @param date The date to format.
-     * @param format The format string to use.
-     * @param locale The locale to use.
-     */
-    public TextOf(
-        final OffsetDateTime date,
-        final String format,
-        final Locale locale
-    ) {
-        this(date, DateTimeFormatter.ofPattern(format, locale));
-    }
-
-    /**
-     * Formats the date using the provided formatter.
-     * @param date The date to format.
-     * @param formatter The formatter to use.
-     */
-    public TextOf(
-        final OffsetDateTime date,
-        final DateTimeFormatter formatter
-    ) {
-        this(() -> formatter.format(date));
-    }
-
-    /**
-     * Formats date using ISO date time format.
-     * @param date The date to format.
-     */
-    public TextOf(final ZonedDateTime date) {
-        this(date, new Iso().value());
-    }
-
-    /**
-     * Formats date using provided date time format string using default locale.
-     * @param date The date to format.
-     * @param format The format to use.
-     */
-    public TextOf(final ZonedDateTime date, final String format) {
-        this(date, format, Locale.getDefault(Locale.Category.FORMAT));
-    }
-
-    /**
-     * Formats the date using the provided format string using the provided
-     * locale.
-     * @param date The date to format.
-     * @param format The format string to use.
-     * @param locale The locale to use.
-     */
-    public TextOf(
-        final ZonedDateTime date,
-        final String format,
-        final Locale locale
-    ) {
-        this(date, DateTimeFormatter.ofPattern(format, locale));
-    }
-
-    /**
-     * Formats the date using the provided formatter.
-     * @param date The date to format.
-     * @param formatter The formatter to use.
-     */
-    public TextOf(
-        final ZonedDateTime date,
-        final DateTimeFormatter formatter
-    ) {
-        this(() -> formatter.format(date));
-    }
-
     /**
      * Ctor.
      *
      * @param scalar The Scalar of String
      */
     public TextOf(final Scalar<String> scalar) {
-        this.origin = scalar;
-    }
-
-    @Override
-    public String asString() throws Exception {
-        return this.origin.value();
+        super(new TextOfScalar(scalar));
```

Comment:
@baudoliver7 the private constructor taking a `Text` was a good idea IMHO, instead of forcing other constructor to go through the use of a `Scalar` (for example the constructor for `Iterable`).

Reply:
@victornoel I agree with you. But by bringing back `Bytes` constructors, there is a conflict between `Scalar` and `Text` constructors.

Reply:
@baudoliver7 we can use `TextOfScalar` directly to define the `Bytes` constructor I think

Reply:
@victornoel Right !